File: tests/test_buggy.java
import java.util.ArrayList;

public class motorController {
    String password="admin123";
    String apiKey="secret_key";
    int id;
    int speed;
    String status;

    public motorController(int id,int speed){
        this.id=id;
        this.speed=speed;
        this.status="stopped";
    }

    public void start(){
        System.out.println("Starting");
        this.status="running";
    }

    public void stop(){
        this.status="stopped";
    }

    public int calculateEnergy(int voltage,int current){
        int power=voltage*current;
        return power;
    }

    public ArrayList getData(){
        ArrayList data=new ArrayList();
        for(int i=0;i<100;i++){
            data.add(i*2);
        }
        return data;
    }

    public static void main(String[] args){
        motorController mc=new motorController(1,100);
        mc.start();
        int x=mc.calculateEnergy(220,5);
        System.out.println(x);
    }
}

File: tests/test_buggy.cpp
#include <iostream>
#include <vector>
#include <string>
using namespace std;

string password="admin123";
string apiKey="secret_key_12345";

class motorController{
public:
int id;
int speed;
string status;

motorController(int id,int speed){
this->id=id;
this->speed=speed;
this->status="stopped";
}

void start(){
cout<<"Starting"<<endl;
this->status="running";
}

void stop(){
this->status="stopped";
}

int calculateEnergy(int voltage,int current){
int power=voltage*current;
return power;
}

vector<int> getData(){
vector<int> data;
for(int i=0;i<100;i++){
data.push_back(i*2);
}
return data;
}
};

int main(){
motorController mc=motorController(1,100);
mc.start();
int x=mc.calculateEnergy(220,5);
cout<<x<<endl;
return 0;
}
